teloxide_tests/src/mock_bot.rs

```diff
@@ -346,6 +319,7 @@ impl MockBot {
         {
             left_tries -= 1;
             if left_tries == 0 {
+                cancel_token.cancel();
                 self.close_bot().await;
```

Comment:
Why no `server.await.unwrap();` here?

Reply:
Also, i am now wondering, is it possible, using `server`, to check if it has started...

Reply:
honestly, I don't know why this one doesn't need it and the other one does. It seems to work for the moment. #19 should make this all a lot simpler by putting the server cleanup in a drop method.